lib/google_places/spot.rb

```diff
@@ -383,6 +383,33 @@ def self.list_by_query(query, api_key, options = {})
       request(:spots_by_query, multipage_request, exclude, options)
     end
 
+    # Search for a Spot with a query
+    #
+    # @return [Spot]
+    # @param [String] query the query to search for
+    # @param [String] api_key the provided api key
+    # @param [Hash] options
+    # @option options [String] :inputtype (textquery) The type of input to query (textquery or phonenumber)
+    # @option options [Hash] :retry_options ({})
+    #   A Hash containing parameters for search retries
+    # @option options [Object] :retry_options[:status] ([])
+    # @option options [Integer] :retry_options[:max] (0) the maximum retries
+    # @option options [Integer] :retry_options[:delay] (5) the delay between each retry in seconds
+    def self.by_query(query, api_key, options = {})
+      inputtype = options.delete(:inputtype) || 'textquery'
+      retry_options = options.delete(:retry_options) || {}
+
+      options = {
+        :input => query,
+        :inputtype => inputtype,
+        :key => api_key,
+        :retry_options => retry_options,
+        :results_key => 'candidates'
+      }
+
+      request(:spot_by_query, false, [], options).first
```

Comment:
Should probably move this `first` call to the `Request` class so the request always returns a single `Spot`.